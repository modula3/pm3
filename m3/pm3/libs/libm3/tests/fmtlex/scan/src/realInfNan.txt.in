# Last modified on Sun Feb 27 16:51:01 PST 1994 by heydon
#      modified on Fri Feb 25 15:00:16 PST 1994 by kalsow
#
# This file tests that the Lex implemenation parses NaN's and infinities
# correctly on IEEE implementations.

# single-precision
r inf		# =>  Infinity
r -inf		# => -Infinity
r INFINITY	# =>  Infinity
r -INFINITY	# => -Infinity
r nan		# => NaN
r NaN		# => NaN

# double-precision
l inf		# =>  Infinity
l -inf		# => -Infinity
l INFINITY	# =>  Infinity
l -INFINITY	# => -Infinity
l nan		# => NaN
l NaN		# => NaN

# extended-precision
e inf		# =>  Infinity
e -inf		# => -Infinity
e INFINITY	# =>  Infinity
e -INFINITY	# => -Infinity
e nan		# => NaN
e NaN		# => NaN

# lex errors
r in		# => Lexical Error
r infi		# => Lexical Error
r na		# => Lexical Error

