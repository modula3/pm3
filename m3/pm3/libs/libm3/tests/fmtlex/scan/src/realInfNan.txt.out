r inf		# =>  Infinity
-> Infinity

r -inf		# => -Infinity
-> -Infinity

r INFINITY	# =>  Infinity
-> Infinity

r -INFINITY	# => -Infinity
-> -Infinity

r nan		# => NaN
-> NaN

r NaN		# => NaN
-> NaN

l inf		# =>  Infinity
-> Infinity

l -inf		# => -Infinity
-> -Infinity

l INFINITY	# =>  Infinity
-> Infinity

l -INFINITY	# => -Infinity
-> -Infinity

l nan		# => NaN
-> NaN

l NaN		# => NaN
-> NaN

e inf		# =>  Infinity
-> Infinity

e -inf		# => -Infinity
-> -Infinity

e INFINITY	# =>  Infinity
-> Infinity

e -INFINITY	# => -Infinity
-> -Infinity

e nan		# => NaN
-> NaN

e NaN		# => NaN
-> NaN

r in		# => Lexical Error
** Raised "Lex.Error"

r infi		# => Lexical Error
** Raised "Lex.Error"

r na		# => Lexical Error
** Raised "Lex.Error"

